File: apps/web/src/code/js/basics/fs_watch.js
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({ template: 'base' })

// Start filesystem watcher for the /home directory
const watcher = sandbox.filesystem.watchDir('/home') // $HighlightLine
watcher.addEventListener((event) => {
  // $HighlightLine
  console.log('Filesystem event', event) // $HighlightLine
}) // $HighlightLine
await watcher.start() // $HighlightLine

// Create files in the /home directory inside the playground
// We'll receive notifications for these events through the watcher we created above.
for (let i = 0; i < 10; i++) {
  console.log('Creating file', i)
  // `filesystem.write()` will trigger two events:
  // 1. 'Create' when the file is created
  // 2. 'Write' when the file is written to
  await sandbox.filesystem.write(`/home/${i}.txt`, `Hello World ${i}!`)
}

await sandbox.close()

File: apps/web/src/code/python/basics/fs_watch.py
import time

from e2b import Sandbox

sandbox = Sandbox.create()

watcher = sandbox.files.watch_dir("/home")  # $HighlightLine

# Create files in the /home directory inside the playground
# We'll receive notifications for these events through the watcher we created above.
for i in range(10):
    # `filesystem.write()` will trigger two events:
    # 1. 'Create' when the file is created
    # 2. 'Write' when the file is written to
    sandbox.files.write(f"/home/file{i}.txt", f"Hello World {i}!")
    time.sleep(1)

for event in watcher.get():
    print(f"Event: {event.type} {event.name}")  # $HighlightLine

watcher.stop()
sandbox.kill()
